net/etesync-server/files/80_setup-etesync-server

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+[ -x /etc/init.d/etesync-server ] && /etc/init.d/etesync-server stop
+
+cd /usr/share/etesync-server || exit 1
+
+if [ ! -f "db.sqlite3" ]; then
+    echo "etesync-server: initializing db.sqlite3 ..."
+    python3 manage.pyc migrate --noinput || exit 1
+    python3 manage.pyc shell -c "from django.contrib.auth.models import User; \
+        User.objects.create_superuser('root', '', 'root')"  || exit 1
+    echo "etesync-server: created user 'root' with password 'root'."
```

Comment:
Yes, this password should be changed (this login is for creating users). Is there a standard way for setup?

Reply:
I don't think there is a standard way to do setup - something like debconf would be helpful here.

If there is an easy way for users to change this root password later (i.e. through a web interface), I guess it's ok to set a default password during installation (with the understanding that some users may never change the default). Large warnings to tell the user to change the default password would also help :joy:

Reply:
I am doing it now with user interaction in this file. Without TTY it is skipped ...

Reply:
I think the best would be to force the user to change the password (making it `unusable` at install), I will ask @tasn about it, maybe he knows if this can be done.

Reply:
As far as I know, there's no way to force a password change in django without adding extra dependencies and code. Though you probably don't want to go down that path anyway.
What about generating a random password here and printing it for the user to save? Or alternatively as @jefferyto suggested, have them set it up manually?

I can try and think about something tomorrow, but no better ideas come to mind at the moment. What users usually do is just create the superuser manually after installation. Maybe that's another option to consider?

Reply:
Thank you very much, I think the best will then be to let the user set it up :-)

Reply:
Maybe print out a message about it. All the user needs to do is: `/path/to/manage.py createsuperuser` and they will then be prompted with questions.
Maybe we should add a management command to let users create normal users directly so they don't need to even create a superuser (or use the web admin). Just a thought. What do you think?

Reply:
Yes printing the message is good, I will look into that.

I am not sure if I understood you right (management command=manage.py?), I will think about it, but I think this is not really needed. Maybe it would be a nice to have in a larger setup, but on a router it is good if there is a superuser who can create accounts for the family ;-)

Reply:
True, you are right. It's a good idea to create a superuser account. Nevermind me. :)